tensorflow/core/ops/training_ops.cc

```diff
@@ -102,6 +102,30 @@ use_locking: If `True`, the subtraction will be protected by a lock;
   otherwise the behavior is undefined, but may exhibit less contention.
 )doc");
 
+REGISTER_OP("ApplyDelayCompensatedGradientDescent")
```

Comment:
Did you also want me to rename the op to "ResourceApplyDelayCompensatedGradientDescent"?